lib/lpc55-rot-startup/src/images.rs

```diff
@@ -120,20 +120,26 @@ impl Image {
         return true;
     }
 
-    // TODO: This is a particularly naive way to calculate the hash of the
-    // hubris image: https://github.com/oxidecomputer/hubris/issues/736
-    pub fn get_hash(&self) -> [u8; 32] {
-        let img_ptr = self.get_img_start() as *const u8;
-        // The MPU requires 32 byte alignment and so the compiler pads the
-        // image accordingly. The length field from the image header does not
-        // (and should not) account for this padding so we must do that here.
-        let img_size = self.get_img_size().unwrap_lite() + 31 & !31;
-        let image = unsafe { core::slice::from_raw_parts(img_ptr, img_size) };
-
-        let mut img_hash = Sha3_256::new();
-        img_hash.update(image);
+    pub fn get_fwid(&self) -> [u8; 32] {
+        let mut hash = Sha3_256::new();
+
+        for start in IMAGE_FLASH.step_by(FLASH_PAGE_SIZE) {
```

Comment:
Okay while testing I realize this doesn't actually work. We need to be able to get the base of the other image, otherwise they both end up with the same hash:

```
$ humility -a target/rot-carrier/dist/a/build-rot-carrier-image-a.zip hiffy --call Update.rot_boot_info
humility: attached via CMSIS-DAP
Update.rot_boot_info() => RotBootInfo { active: A, persistent_boot_preference: A, pending_persistent_boot_preference: None, transient_boot_preference: None, slot_a_sha3_256_digest: Some([ 0x9d, 0xce, 0xc5, 0x7e, 0x78, 0xb8, 0x74, 0xb0, 0xc8, 0x78, 0x71, 0x7e, 0xf1, 0x58, 0x8b, 0x53, 0x88, 0xea, 0xc, 0x7, 0xb6, 0x23, 0x3, 0x69, 0x3b, 0x8f, 0xd4, 0xd8, 0x1d, 0x10, 0x8f, 0x1 ]), slot_b_sha3_256_digest: Some([ 0x9d, 0xce, 0xc5, 0x7e, 0x78, 0xb8, 0x74, 0xb0, 0xc8, 0x78, 0x71, 0x7e, 0xf1, 0x58, 0x8b, 0x53, 0x88, 0xea, 0xc, 0x7, 0xb6, 0x23, 0x3, 0x69, 0x3b, 0x8f, 0xd4, 0xd8, 0x1d, 0x10, 0x8f, 0x1 ]) }
```

I'll give this a think if there's another option

Reply:
I see what you mean. This doesn't become apparent until you've flashed the 'b' image before pulling the `rot_boot_info`:

```
(hubris) [flihp@slag hubris]$ humility --target rot-carrier --archive-name imagea -- hiffy --call Update.rot_boot_info
humility: attached to 1fc9:0143:IYJ1NYJRCW2TC via CMSIS-DAP
Update.rot_boot_info() => RotBootInfo { active: B, persistent_boot_preference: A, pending_persistent_boot_preference: None, transient_boot_preference: None, slot_a_sha3_256_digest: Some([ 0x3d, 0x4d, 0x34, 0x69, 0x37, 0x24, 0x8b, 0xef, 0x93, 0xd7, 0x18, 0xd1, 0x25, 0xf3, 0xe0, 0xab, 0x15, 0xbc, 0x15, 0x98, 0x33, 0xaa, 0x46, 0x67, 0x91, 0xa4, 0x26, 0x53, 0xde, 0x53, 0x10, 0x41 ]), slot_b_sha3_256_digest: Some([ 0x3d, 0x4d, 0x34, 0x69, 0x37, 0x24, 0x8b, 0xef, 0x93, 0xd7, 0x18, 0xd1, 0x25, 0xf3, 0xe0, 0xab, 0x15, 0xbc, 0x15, 0x98, 0x33, 0xaa, 0x46, 0x67, 0x91, 0xa4, 0x26, 0x53, 0xde, 0x53, 0x10, 0x41 ]) }
```

It's pretty much impossible for the two images to have the same hash (and still be functional) so this is definitely wrong. I'll sort out some options and update the PR.

Reply:
I've just pushed two commits with a new approach. Instead of exporting the range for the image being built we now export all flash ranges for the chip from xtask to through the environment. This allows us to use the appropriate range for A or B images. This also duplicates data in the `HUBRIS_DICE_MFG` env var so I've included another commit removing it.